flake.nix

```diff
@@ -12,92 +12,220 @@
+                ${lib.optionalString (cfg.tailscaleAuthKeyFile != null) ''
+                  export TS_AUTHKEY="$(head -n1 ${lib.escapeShellArg cfg.tailscaleAuthKeyFile})"
+                ''}
+                ${cfg.package}/bin/tclipd ${builtins.concatStringsSep " " args};
+              '';
+              wantedBy = ["multi-user.target"];
+              serviceConfig = {
+                User = cfg.user;
+                Group = cfg.group;
+                Restart = "always";
+                RestartSec = "15";
+                WorkingDirectory = "${cfg.dataDir}";
+              };
+            };
+          };
+        };
+
+      nixosModules.default = self.nixosModules.tclip;
+    }
+    // utils.lib.eachSystem [
       "x86_64-linux"
       "aarch64-linux"
       "x86_64-darwin"
       "aarch64-darwin"
-    ] (system:
-      let
-      graft = pkgs: pkg: pkg.override {
-        buildGoModule = pkgs.buildGo122Module;
-      };
-        pkgs = import nixpkgs {
-          inherit system;
-          overlays = [ gomod2nix.overlays.default (final: prev: {
+    ] (system: let
+      graft = pkgs: pkg:
+        pkg.override {
+          buildGoModule = pkgs.buildGo122Module;
+        };
+      pkgs = import nixpkgs {
+        inherit system;
+        overlays = [
+          gomod2nix.overlays.default
+          (final: prev: {
             go = prev.go;
             go-tools = graft prev prev.go-tools;
             gotools = graft prev prev.gotools;
             gopls = graft prev prev.gopls;
-          }) ];
+          })
+        ];
+      };
+      version = builtins.substring 0 8 self.lastModifiedDate;
+    in {
+      # Formatter for your nix files, available through 'nix fmt'
+      # Other options beside 'alejandra' include 'nixpkgs-fmt'
+      formatter = pkgs.alejandra;
+
+      packages = rec {
+        tclipd = pkgs.buildGoApplication {
+          pname = "tclipd";
+          version = "0.1.0-${version}";
+          go = pkgs.go;
+          src = ./.;
+          subPackages = "cmd/tclipd";
+          modules = ./gomod2nix.toml;
         };
-        version = builtins.substring 0 8 self.lastModifiedDate;
-      in {
-        packages = rec {
-          tclipd = pkgs.buildGoApplication {
-            pname = "tclipd";
-            version = "0.1.0-${version}";
-            go = pkgs.go;
-            src = ./.;
-            subPackages = "cmd/tclipd";
-            modules = ./gomod2nix.toml;
-          };
 
-          tclip = pkgs.buildGoApplication {
-            pname = "tclip";
-            inherit (tclipd) src version modules;
-            subPackages = "cmd/tclip";
-            go = pkgs.go;
+        tclip = pkgs.buildGoApplication {
+          pname = "tclip";
+          inherit (tclipd) src version modules;
+          subPackages = "cmd/tclip";
+          go = pkgs.go;
 
-            CGO_ENABLED = "0";
-          };
+          CGO_ENABLED = "0";
+        };
 
-          docker = pkgs.dockerTools.buildLayeredImage {
```

Comment:
These changes are from formatting; I added `formatter` as an attribute so `nix fmt` should keep this consistent.